src/utils/scale-tick-intervals.js

```diff
@@ -1,142 +1,135 @@
-
 /**
  * Code to calculate which values to label in a scale between two
  * endpoints, and how they should be labelled. Based on
  * ScaleTickIntervals from Sonic Visualiser, relicensed for this
  * library. Copyright 2017 QMUL.
  */
 
-export default class ScaleTickIntervals {
-
-  constructor() { }
-  
-  /**
-   * Return an array of objects describing tick locations and labels,
-   * each object having "value" (number) and "label" (string)
-   * properties. All ticks will be within the range [min, max] and
-   * there will be approximately n+1 of them, dividing the range up
-   * into n divisions, although this number may vary based on which
-   * tick values seem best suited to labelling.
-   */
-  linear(min, max, n) {
-    let instruction = this._linearInstruction(min, max, n);
-    return this._explode(instruction);
+/**
+ * Return an array of objects describing tick locations and labels,
+ * each object having "value" (number) and "label" (string)
+ * properties. All ticks will be within the range [min, max] and
+ * there will be approximately n+1 of them, dividing the range up
+ * into n divisions, although this number may vary based on which
+ * tick values seem best suited to labelling.
+ */
+export function linear(min, max, n) {
+  return explode(linearInstruction(min, max, n));
+}
+
+function linearInstruction(min, max, n) {
+  let display = "auto";
+  if (max < min) {
+    return linearInstruction(max, min, n);
+  }
+  if (n < 1 || max === min) {
+    return {
+      initial: min, limit: min, spacing: 1.0,
+      roundTo: min, display, precision: 1, logUnmap: false
+    };
+  }
+  if (min !== min || max !== max) {
+    // NaNs must be involved
+    console.log("ScaleTickIntervals: WARNING: min = " + min + ", max = " + max);
+    return [];
   }
 
```

Comment:
I assume the vast majority of changes below here are just re-indentation -- I haven't checked